pkg/aws/ec2/api/helper.go

```diff
@@ -594,3 +619,81 @@ func (h *ec2APIHelper) DetachAndDeleteNetworkInterface(attachmentID *string, nwI
 	}
 	return nil
 }
+
+func (h *ec2APIHelper) GetSecurityGroupIdsForSecurityGroupNames(securityGroupNames []string) ([]string, error) {
+	sgIds := make([]string, 0)
+	if len(securityGroupNames) < 1 {
+		return sgIds, nil
+	}
+	sgNamesNotInCache := make([]string, 0)
+	for _, sgName := range securityGroupNames {
+		if sgId, ok := h.securityGroupNameToIdMap[sgName]; !ok {
```

Comment:
I'm not sure caching is a good idea here. 

Consider the scenario where a security group policy includes `sg-example-name` corresponding to `sg-1111`. This is added to `securityGroupNameToIdMap`, and some pod A is launched with `sg-1111` attached to the branchENI. Now, customer deleted `sg-1111` and created `sg-2222` with the same name `sg-example-name`. But the map still points to old `sg-1111` which no longer exists. So any pods launched after would be stuck in creation process as branchENI cannot be created(due to missing SG). 

Need to evaluate if we always have to do an API call during pod creation or we can consider periodically refreshing the cache (TTL caching).

Reply:
I think a TTL caching make sense given (at least from our side) how rare we change delete security group and recreate with the same name. The cache was added to reduce the number of calls we hit EC2 API with.